### PR TITLE
test backend post schema update

### PR DESCRIPTION
**Pull request by lwly-jpg**
('Testing' as in playing around with, no actual Jest test for this, just wanted to explore schema model a little!)

Changes only to backend PostSchema (Model) - added:

- 'required' for message (all posts must have a message, so no blank posts are stored in DB)
- likes, with a default of 0 // NOTE: @mixxxxxer's suggestion on using an Array here vs String
- image (which can take a URL as a string) 
- timestamp (logs date and time of post creation)

Came across this exploring how we can add comments:
https://stackoverflow.com/questions/65931572/node-js-mongoose-create-a-blog-post-commenting-system

TLDR: We could create a separate CommentSchema later on e.g. (code block loses formatting but just to illustrate!):

```
const CommentSchema = new mongoose.Schema({
         message: { type: String, trim: true, required: true },
       // each comment only relates to one post
        post: { type: mongoose.Schema.Types.ObjectId, ref: 'Post' }
         }, {timestamps: true})
```

And add the below to the PostSchema (stores comments in an array in a post document):

> `    comments: [{
>              type: mongoose.Schema.Types.ObjectId,
>              ref: 'Comment'
>            }]`

### CODE REVIEW

**Review by josephclander — APPROVED** [review]
I think so while we think about it.
Then additionally to this code, in the user model you can get the other information like a SQL join with `.populate()`

**Review by mixxxxxer — APPROVED** [review]

File: api/models/post.js [hunk 1]
```diff
@@ -9,8 +9,20 @@ const mongoose = require("mongoose");
...
 const PostSchema = new mongoose.Schema({
-  message: String
-});
+  message: {
+    type: String,
+    required: true
+  },
+  likes: {
+    type: Number,
+    default: 0,
```
[the remark this answers is not in the document]
Comment (lwly-jpg) [reply api/models/post.js:18]: Sounds great, thank you @mixxxxxer for the suggestion!

**Review by mixxxxxer — APPROVED** [review]
Yes, I believe it's will work!)

**Review by josephclander — APPROVED** [review]
I see its a number in an array now, so Id in an array will now work.
Great.
